packages/dart_frog_cli/lib/src/daemon/daemon_server.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:async';
+
+import 'package:dart_frog_cli/src/daemon/daemon.dart';
+import 'package:mason/mason.dart';
+import 'package:meta/meta.dart';
+
+/// The version of the Dart Frog daemon protocol.
+///
+/// This exists so clients can know if they are compatible with the
+/// format and order of the events, the methods,
+/// its arguments and return values.
+///
+/// This does not have to follow the version of the
+/// package containing this class.
+const daemonVersion = '0.0.1';
+
+/// {@template daemon}
+/// The Dart Frog daemon.
+///
+/// The daemon is a persistent routine that runs, analyzes and manages
+/// Dart Frog projects.
+/// {@endtemplate}
+class DaemonServer {
+  /// {@macro daemon}
+  DaemonServer({
+    @visibleForTesting DaemonConnection? connection,
+  }) : _connection = connection ?? DaemonStdioConnection() {
+    _connection.inputStream.listen(_handleMessage);
+
+    addDomain(DaemonDomain(this));
+  }
+
+  final Map<String, Domain> _domains = {};
+
+  final DaemonConnection _connection;
+
+  final _exitCodeCompleter = Completer<ExitCode>();
+
+  /// A [Future] that completes when the daemon exits.
+  Future<ExitCode> get exitCode => _exitCodeCompleter.future;
+
+  /// The names of the domains in the daemon.
+  Iterable<String> get domainNames => _domains.keys;
+
+  /// Whether the daemon has exited.
+  bool get isCompleted => _exitCodeCompleter.isCompleted;
+
+  /// The version of the Dart Frog daemon protocol.
+  String get version => daemonVersion;
+
+  /// Adds a [domain] to the daemon.
+  ///
+  /// Visible for testing purposes only.
```

Comment:
Nit: already explicit with the annotation.

```suggestion
```

Reply:
The annotation looks quite obscure on dart doc generated pages. That's why I always like to add this to the dardoc.